main.py

```diff
@@ -47,60 +61,76 @@ def check_if_buy(isin: str, x1: str = "d1"):
-            time.sleep(14400)  # check back in 4 hours
         except Exception as e:
             print(f'1{e}')
-            time.sleep(60)
     else:
         try:
             # create a sell order if mean reversion decision returns False
             print('sell')
-            placed_order = Order(
+            placed_order = client.trading.orders.create(
                 isin=isin,
                 expires_at="p7d",
                 side="sell",
                 quantity=1,
                 venue=os.getenv("MIC"),
-            ).place_order().get('results', None)
+            )
             # if position in portfolio, activate order
             if placed_order is not None:
-                order_id = placed_order.get('id')
-                activated_order = Order().activate_order(order_id)
+                order_id = placed_order.results.id
+                activated_order = client.trading.orders.activate(order_id)
                 print(activated_order)
             else:
                 print("You do not have sufficient holdings to place this order.")
-            time.sleep(14400)  # check back in 4 hours
         except Exception as e:
             print(f'2{e}')
-            time.sleep(60)
 
 
-def mean_reversion():
-    """
-    main function to be executed
-    """
-    while True:
-        if TradingVenue().check_if_open():
-            # make buy or sell decision
-            check_if_buy(
-                isin="US88160R1014",  # this is Tesla, but you can obviously use any ISIN you like :)
-                x1="d1"
-            )
-        else:
-            # sleep until market reopens in case it is closed
-            time.sleep(TradingVenue().seconds_till_tv_opens())
+def schedule_trades_for_year():
+    opening_days = client.market_data.venues.get().results[0].opening_days
+
+    for day in opening_days:
+        # using venues endpoint we only schedule trades for days the exchange is open
+        date_string = f'{day:%Y-%m-%d}'
+        year_month_day = date_string.split('-')
+        for x in range(13):
+            # using a scheduler, we run the mean reversion logic once per hour starting from 8:30
+            # (based on Munich Stock Exchange hours)
+            scheduler.add_job(mean_reversion,
+                              trigger=CronTrigger(year=year_month_day[0],
+                                                  month=year_month_day[1],
+                                                  day=year_month_day[2],
+                                                  hour=8 + x,
+                                                  minute=30,
+                                                  timezone=utc),
+                              name="Perform Mean Reversion Hourly")
 
 
 if __name__ == '__main__':
-    mean_reversion()
+    scheduler = BlockingScheduler(timezone=utc)
+
+    # reschedule your trades for the next years ad infinitum
+    schedule_trades_for_year()
+    scheduler.add_job(schedule_trades_for_year,
```

Comment:
Currently schedules trades for all days left in current year (line 123). Then adds a job which will schedule trades for all upcoming years on New Years Day (line 124-129).